src/main/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessor.java

```diff
@@ -0,0 +1,274 @@
+    @Autowired
+    public final void setDynamoHelper(DynamoHelper dynamoHelper) {
+        this.dynamoHelper = dynamoHelper;
+    }
+
+    @Autowired
+    public final void setZipHelper(ZipHelper zipHelper) {
+        this.zipHelper = zipHelper;
+    }
+
+    public final void setPageSize(int pageSize) {
+        this.pageSize = pageSize;
+    }
+
+    /** Main entry point into Download Participant Roster Worker. */
+    @Override
+    public void accept(JsonNode jsonNode) throws Exception {
+
+        DownloadParticipantRosterRequest request;
+        try {
+            request = DefaultObjectMapper.INSTANCE.treeToValue(jsonNode, DownloadParticipantRosterRequest.class);
+        } catch (IOException e) {
+            throw new PollSqsWorkerBadRequestException("Error parsing request: " + e.getMessage(), e);
+        }
+
+        String userId = request.getUserId();
+        String appId = request.getAppId();
+        String password = request.getPassword();
+        String studyId = request.getStudyId();
+
+        LOG.info("Received request for userId=" + userId + ", app=" + appId + ", studyId=" + studyId);
+
+        Stopwatch requestStopwatch = Stopwatch.createStarted();
+        File csvFile = null;
+        File tmpDir = fileHelper.createTempDir();
+        File zipFile = null;
+
+        try {
+            // get caller's user
+            StudyParticipant participant = bridgeHelper.getParticipant(appId, userId, false);
+            String orgMembership = participant.getOrgMembership();
+            List<Role> participantRoles = participant.getRoles();
+
+            if (participantRoles == null || participantRoles.isEmpty() ||
+                    !(participantRoles.contains(Role.RESEARCHER) || participantRoles.contains(Role.STUDY_COORDINATOR))) {
+                LOG.info("User does not have a Researcher or Study Coordinator role.");
+                return;
+            }
+
+            if (participantRoles.contains(Role.STUDY_COORDINATOR) && orgMembership == null) {
+                LOG.info("User is a Study Coordinator and does not have an org membership.");
+                return;
+            }
+
+            if (studyId != null && participantRoles.contains(Role.STUDY_COORDINATOR)) {
+
+                List<Study> studies = bridgeHelper.getStudiesForApp(appId, orgMembership, 0, pageSize);
+                if (studies.isEmpty()) {
+                    LOG.info("User's org does not sponsor the given studyId.");
+                    return;
+                }
+            }
+
+            if (participant.getEmail() == null || !participant.isEmailVerified()) {
+                LOG.info("User does not have a validated email address.");
+                return;
+            }
+
+            int offsetBy = 0;
+            // get first page of account summaries
+            List<AccountSummary> accountSummaries = bridgeHelper.getAccountSummariesForApp(appId, orgMembership,
+                    offsetBy, pageSize, studyId);
```

Comment:
Also, Alx mentioned that the Account Summary doesn't contain all fields, so we need to call getParticipantByIdForApp() to get all the Account fields.